File: ChopraSSO/Classes/ChopraSSO.m
//
//  ChopraSSO.m
//  Pods
//
//  Created by Örs Kovács on 2017. 03. 20..
//
//

#import <Foundation/Foundation.h>

#import "ChopraSSO.h"
#import "FacebookHelper.h"
#import "GoogleHelper.h"

@implementation ChopraSSO

static NSString* _googleClientID;

+(void)setGoogleClientID:(NSString *)googleClientID {
    _googleClientID = googleClientID;
    [[GoogleHelper sharedInstance] setClientID:googleClientID];
}

+(NSString *)googleClientID {
    return _googleClientID;
}

+ (BOOL)application:(UIApplication *)app
            openURL:(NSURL *)url
            options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
    return [[FacebookHelper sharedInstance] application:app openURL:url options:options] ||
            [[GoogleHelper sharedInstance] application:app openURL:url options:options];
}

+ (BOOL)application:(UIApplication *)application
            openURL:(NSURL *)url
            sourceApplication:(NSString *)sourceApplication
            annotation:(id)annotation {
    return [[FacebookHelper sharedInstance] application:application openURL:url sourceApplication:sourceApplication annotation:annotation] ||
            [[GoogleHelper sharedInstance] application:application openURL:url sourceApplication:sourceApplication annotation:annotation];
}

@end

File: ChopraSSO/Classes/ChopraSSO.swift
//
//  ChopraSSO.swift
//  ChopraSSO
//
//  Created by Örs Kovács on 2018. 06. 22..
//

import Foundation

public final class ChopraSSO {
    public static var googleClientId: String? {
        return GoogleHelper.shared.clientId
    }
    
    public static func setGoogleClientID(_ newId: String?) {
        GoogleHelper.shared.clientId = newId
    }
    
    public static func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
        return GoogleHelper.shared.application(app, open: url, options: options)
            || FacebookHelper.shared.application(app, open: url, options: options)
    }
    
    public static func application(_ application: UIApplication, open url: URL, sourceApplication: String?, annotation: Any) -> Bool {
        return GoogleHelper.shared.application(application, open: url, sourceApplication: sourceApplication, annotation: annotation)
            || FacebookHelper.shared.application(application, open: url, sourceApplication: sourceApplication, annotation: annotation)
    }
}
